File: app/src/main/java/com/gazilla/mihail/gazillaj/presentation/main/card/CardView.java
package com.gazilla.mihail.gazillaj.presentation.main.card;

import android.graphics.Bitmap;

import java.util.Map;

public interface CardView {

    void setValueProgressBar(int maxValue, int userValue);
    void initListWithLvl(int myLvl, Map<Integer, Integer> mapLvl);
    void setQRcode(Bitmap bitmap);
    void setSpins(int qty);
    void myWin(String win, String res);
    void initLvlForKoleso(int res);
    void startWheeling();
    void firstDialogTip();
    void wheelTip(Boolean show);
    void balanceTip(Boolean show);
    void nacopTip(Boolean show);
    void lvlDraconTip(Boolean show);
    void registrTip(Boolean show);
    void reserveTip(Boolean show);
    void nextTip(int N);
}

File: app/src/main/java/com/gazilla/mihail/gazillaj/views/CardView.kt
package com.gazilla.mihail.gazillaj.views

import android.graphics.Bitmap
import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(value = AddToEndSingleStrategy::class)
interface CardView:MvpView {

    fun setValueProgressBar(maxValue: Int, userValue: Int)
    fun setQrCode(bitmap: Bitmap)
    fun setSpins(qty: Int)
    fun showWhiteRoundDrakonTip(show: Boolean)
    fun setLvlUserDraconAdapter(levels: Map<Int, Int>)

    fun initWheelLvl(res: Int)

    fun startWheeling()
    fun showMyWin(res: Int, description: String)
    fun showDetailLvlDracon(myLvlDiscription: String, one: String, two: String, fri: String, four: String, fif : String)

    fun updateBalance()

    fun openReserveActivity()

    fun openMessageWithReserve(message: String)
    fun showMessageDialog(message: String)
}